s3pd/__init__.py

```diff
@@ -103,6 +104,38 @@ def download_chunk(bucket, key, shmfileno, offset_first, offset_last, signed):
             Range='bytes=%s-%s' % (offset_first, offset_last))['Body']
         chunk._raw_stream.readinto(shmmap)
 
+def resolve_link(bucket, key, signed, depth=10):
+    # Stop after too many link indirections
+    assert depth > 0, 'Too many levels of link indirections'
+
+    client = create_client(signed)
+    filesize = get_filesize(client, bucket, key)
+
+    # There is no need to resolve files with a size >1KB, these could not
+    # realistically be links
+    if filesize > 1024:
+        return bucket, key
+
+    link_sentinel = '#S3PDLINK#'
```

Comment:
This looks a bit hackish to invent the concept of S3 links
=> So I likea the idea very much.

Could you rename it to #S3LINK# instead of #S3PDLINK# ?
So that we make sure no matter what library we use later on, "#S3LINK#" still makes sense.
This might be moved to a more low-level library later on.

Reply:
Disagreed: we are not AWS nor the boto3 maintainers. We implement that magic in s3pd, so IMO it makes sense to namespace the sentinel with "S3PD".

Reply:
I was originally on the opinion of @pmuller , which is why I used `#S3PDLINK#`, but now that I think of it, if we drop support of `s3pd` for something else in the future, it would make sense to add link support to it, and thus have a more generic sentinel.
What do you guys think?

Reply:
Yes exactly, those links are very generic and not linked only to S3PD.

If we deprecate S3PD later on, we'll still want to keep those links. 
If a library we build does not support S3PD but another (more generic) library, we'll still to use those links.

Reply:
In that case, this piece of code shouldn't even live in s3pd. ;-)